pallets/task/src/lib.rs

```diff
@@ -432,10 +507,44 @@ pub mod pallet {
 			Ok(())
 		}
 
+		// Task can be rejected by the creator, which places the task back into progress.
+		pub fn reject_completed_task(task_initiator: &T::AccountId, task_id: &T::Hash) -> Result<(), DispatchError> {
+			
+			// Check if task exists
+			let mut task = Self::tasks(&task_id).ok_or(<Error<T>>::TaskNotExist)?;
+
+			// Check if the owner is the one who created task
+			ensure!(Self::is_task_initiator(task_id, task_initiator)?, <Error<T>>::OnlyInitiatorAcceptsTask);
+
+			// Remove from ownership of initiator
+			<TasksOwned<T>>::try_mutate(&task_initiator, |owned| {
```

Comment:
this can be already be covered if we allow `reject_completed_task` only on task which has `marked_finished()` executed.